### ISSUE TITLE
Differentiate # (Windows) button key to act on specific screen only

### ISSUE THREAD

**Issue by nanangarsyad** [fuhsjr00/bug.n#136]
I wanted to implemented the feature that can differentiate which windows button I pressed.
For example, I wanted to know whether left-windows or right-windows being pressed.
My point is, I'm trying to create a feature where if I pressed left side modifier button (alt, shift, ctrl, win) it'll only applied on specific monitor (not virtual monitor, but multi-monitor).
e.g.
I have 2 physical monitor, in which case the feature I proposed will work like this:
left-win + t -> will changing the layout type to 'tile layout' for my main (1st) screen
left-win + 3 -> will show 3rd virtual desktop on main (1st) screen

right-win + -> will changing the layout type to 'tile layout' for my second screen
right-win + 3 -> will show 3rd virtual desktop on second screen 

I'm a newbie on autohotkey, but more ore less I have some skill on java and python, and just started with c++ and windows programming.
please give me some pointer, where should I start first, so I could implement this feature.
Thank you, and if my wording not clear enough please let me know :)

**Comment by joten** [comment]
You may use `LWin` and `RWin` instead of `#` to configure hotkeys (see the topic [Hotkeys in the AutoHotkey documentation](https://autohotkey.com/docs/Hotkeys.htm)). For your first example you will need an altered `View_setLayout` function; a possible hack of Config.ahk for the first part would be as follows:
```
LWin & t::
  If Not (1 = View_#1_#%Monitor_#1_aView_#1%_layout_#1) {
    View_#1_#%Monitor_#1_aView_#1%_layout_#2 := View_#1_#%Monitor_#1_aView_#1%_layout_#1
    View_#1_#%Monitor_#1_aView_#1%_layout_#1 := 1
  }
  View_arrange(1, Monitor_#1_aView_#1, True)
Return
```
I already shortened the code, but the point is to change `Manager_aMonitor` to 1 (`LWin`) and 2 (`RWin`) -- but do not set the variable `Manager_aMonitor` itself.
As an alternative you may change the active monitor and apply the function afterwards with no need to alter the function itself; the following should work (I did not test it):
```
LWin & t::
  Manager_activateMonitor(1)
  View_setLayout(1)
Return
```

For your second example you would need an altered `Monitor_activateView` function with `aMonitor` set to the respective values, or the following:
```
LWin & 3::
  Manager_activateMonitor(1)
  Monitor_activateView(3)
Return
```

**Comment by larrxu** [comment]
Have you guys thought about following the pattern of e.g. i3-wm, to assign unique view ids across multiple monitors, with each monitor starts with 4 views instead of 10.
```
Monitor1: view 1, 2, 3, 4
Monitor2: view 5, 6, 7, 8
Monitor3: ...
```
This way the user doesn't need to mess with manually mapping views to monitors and the config file continues to work across multiple monitors, e.g., Win+5 will always bring me to second screen's first view. I think this is the most useful function of a window manager (be able to quickly switching between not only windows but also monitors without me dragging the mouse).

**Comment by joten** [comment]
You are right, this does not work in Config.ini. The hack was intended for Config.ahk, which is part of the source. You will have to put it there towards the end of the file as follows:
```
RWin & j::
  Manager_activateMonitor(2)
  View_activateWindow(0, +1)
Return
```

Than [run bug.n](https://github.com/fuhsjr00/bug.n/blob/master/doc/Installing_and_running.md#installing-and-running-bugn) not by starting the executable, but by starting the script Main.ahk with AutoHotkey.
